project-base/storefront/components/Pages/Order/TransportAndPayment/TransportAndPaymentContent.tsx

```diff
@@ -64,10 +64,10 @@ export const TransportAndPaymentContent: FC = () => {
     };
 
     useEffect(() => {
-        if (isCartEmpty) {
+        if (!cart?.items.length) {
             router.replace(cartUrl);
         }
-    }, [isCartEmpty]);
+    }, [cart]);
```

Comment:
If you add the length as dependency instead of the entire cart, you will reduce the number of runs, right?

Reply:
Actually it is not good idea to check length, which can be the same even if items change. So I improved this by adding `cart.items` as dependecy.

Reply:
Okay, correct 👍